File: backend/src/main/java/com/beej/auth/service/AuthService.java
package com.beej.auth.service;

import com.beej.auth.dto.AuthResponseDTO;
import com.beej.auth.dto.LoginRequestDTO;
import com.beej.auth.dto.RegisterRequestDTO;

public interface AuthService {
    
    AuthResponseDTO register(RegisterRequestDTO registerRequest);
    
    AuthResponseDTO login(LoginRequestDTO loginRequest);
    
    AuthResponseDTO refreshToken(String token);
    
    void logout(String token);
    
    void verifyEmail(String token);
    
    void forgotPassword(String email);
    
    void resetPassword(String token, String newPassword);
}

File: frontend/src/services/authService.js
import api from '../api/axios'

const authService = {
  // Login user
  login: async (credentials) => {
    const response = await api.post('/auth/login', credentials)
    return response.data.data
  },

  // Register user
  register: async (userData) => {
    const response = await api.post('/auth/register', userData)
    return response.data.data
  },

  // Logout user
  logout: async () => {
    const response = await api.post('/auth/logout')
    return response.data
  },

  // Get current user
  getCurrentUser: async () => {
    const response = await api.get('/auth/me')
    return response.data.data
  },

  // Refresh token
  refreshToken: async () => {
    const response = await api.post('/auth/refresh')
    return response.data.data
  },

  // Verify email
  verifyEmail: async (token) => {
    const response = await api.get(`/auth/verify/${token}`)
    return response.data
  },

  // Forgot password
  forgotPassword: async (email) => {
    const response = await api.post('/auth/forgot-password', { email })
    return response.data
  },

  // Reset password
  resetPassword: async (token, newPassword) => {
    const response = await api.post('/auth/reset-password', { token, newPassword })
    return response.data
  },
}

export default authService
